### PR TITLE
RA-543 Patient name should not allow leading or trailing spaces 

### PR DESCRIPTION
**Pull request by DADMahindika**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you squash these commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dkayiwa** [comment]
@DADMahindika are you still working on this?

File: api/src/main/java/org/openmrs/PersonName.java [hunk 1]
```diff
@@ -145,22 +146,28 @@ public static PersonName newInstance(PersonName pn) {
 		}
 		PersonName newName = new PersonName(Integer.valueOf(pn.getPersonNameId()));
 		if (pn.getGivenName() != null) {
-			newName.setGivenName(String.valueOf(pn.getGivenName()));
+			String givenNameAftreTrim = pn.getGivenName().trim();
+			newName.setGivenName(String.valueOf(givenNameAftreTrim));
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/PersonName.java]: Why do you have to call String.valueOf?

Comment (DADMahindika) [reply api/src/main/java/org/openmrs/PersonName.java]: 'String.valueOf' part was there in the original code, and i didn't change it. I don't know the exact purpose of that

**Pull request closed without merging**